src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -475,6 +595,22 @@ def load_sub_model(
         # else load from the root directory
         loaded_sub_model = load_method(cached_folder, **loading_kwargs)
 
+    if isinstance(loaded_sub_model, torch.nn.Module) and isinstance(device_map, dict):
+        # remove hooks
+        remove_hook_from_module(loaded_sub_model, recurse=True)
```

Comment:
we have a `remove_all_hooks` method now, maybe just run that during the `DiffusionPipeline.from_pretrained`, as soon as we detect `device_map is not None`
https://github.com/huggingface/diffusers/blob/e49c04d5d667524308cf55d996172c64f1739ae7/src/diffusers/pipelines/pipeline_utils.py#L940

Reply:
Yeah I was aware of it however, we want `recurse=True` to be always the case. But our existing method conditions it differently;

https://github.com/huggingface/diffusers/blob/e49c04d5d667524308cf55d996172c64f1739ae7/src/diffusers/pipelines/pipeline_utils.py#L946

Reply:
I think it would work because the device hook is a `AlignDevicesHook`

maybe we can change the code to this to be more readable (it will not change the behavior)

```python
def remove_all_hooks(self):
        r"""
        Removes all hooks that were added when using `enable_sequential_cpu_offload` or `enable_model_cpu_offload`.
        """
        for _, model in self.components.items():
            if isinstance(model, torch.nn.Module) and hasattr(model, "_hf_hook"):
                is_recurse = isinstance(getattr(model, "_hf_hook"), accelerate.hooks.AlignDevicesHook)
                accelerate.hooks.remove_hook_from_module(model, recurse=is_recurse)
        self._all_hooks = []
```

Reply:
Ah, thanks! Yes, this should work. But there's still a problem. 

In order for `remove_all_hooks()` to work properly and effectively, the underlying pipeline needs to have all the model-level (aka instances of `nn.Module`) loaded. But here, we're removing hooks when loading the individual models. 

Within `remove_all_hooks()`, we rely on `self.components()` which becomes effective _after_ the pipeline constructor has been initialized with all the requested model-level components and others. 

LMK if it's unclear.